src/libFLAC/stream_encoder.c

```diff
@@ -2070,6 +2070,8 @@ FLAC_API const char *FLAC__stream_encoder_get_resolved_state_string(const FLAC__
 	FLAC__ASSERT(0 != encoder->protected_);
 	if(encoder->protected_->state != FLAC__STREAM_ENCODER_VERIFY_DECODER_ERROR)
 		return FLAC__StreamEncoderStateString[encoder->protected_->state];
+	else if(!encoder->private_->verify.decoder)
```

Comment:
It is probably a good idea to add coverage for this to the fuzzers. I've fixed #215 by adding some code to force allocation failures with #419. Apparently, this coverage is lacking, otherwise this change would have turned up sooner